### PR TITLE
Updated javadocs for MailgunEmailClient and EmailClient

### PR DESCRIPTION
**Pull request by enyia21**
#186242068

### CODE REVIEW

File: src/main/java/formflow/library/email/MailgunEmailClient.java [hunk 1]
```diff
@@ -19,9 +19,21 @@
...
+   * String storing the Mailgun domain
+   */
   private final String mailgunDomain;
+  /**
+   * Mailgun messages api object
+   */
   private MailgunMessagesApi mailgunMessagesApi;
+  /**
+   * Boolean variable sets requireTLS to true.
+   */
```
Comment (bseeger) [inline src/main/java/formflow/library/email/MailgunEmailClient.java:36]: Do we need to document private variables? I doubt these show up in the javadocs...?

Comment (bseeger) [reply src/main/java/formflow/library/email/MailgunEmailClient.java:36]: Not that it hurts to document the, but I don't think we _have_ to.

Comment (bseeger) [reply src/main/java/formflow/library/email/MailgunEmailClient.java:36]: Do you have any thoughts on this?  I'm fine keeping it in, just wondering if you knew of a best practice around this?

Comment (enyia21) [reply src/main/java/formflow/library/email/MailgunEmailClient.java]: Documenting private variables is not included in the javadocs warnings for missing comments

File: src/main/java/formflow/library/email/EmailClient.java [hunk 2]
```diff
@@ -3,8 +3,15 @@
...
+/**
+ * Requires sendEmail methods for any class implementing EmailClient.
+ * @param <T> A Generic that can be set as the return type for the sendEmail method
+ */
 public interface EmailClient<T> {
 
+  /**
+   * A boolean variable required for any EmailClient
+   */
   Boolean requireTls = true;
```
Comment (bseeger) [inline src/main/java/formflow/library/email/EmailClient.java:15]: This inspired conversation about EmailClient being an interface and default scope for this variable.  When there is a variable in an Interface, it is by default "public Static final", which is obviously what we don't want here.  We are creating a ticket to rethink the logic here.

Comment (bseeger) [reply src/main/java/formflow/library/email/EmailClient.java:15]: https://www.pivotaltracker.com/story/show/186251813

**Review by bseeger — APPROVED** [review]
Looks good to me!

**Pull request merged**